geoips/plugin_registry.py

```diff
@@ -0,0 +1,294 @@
+                            for subplg in current_registry[plugin_type][
+                                interface
+                            ][plugin]:
+                                self.validate_plugin_attrs(
+                                    plugin_type,
+                                    interface,
+                                    (plugin, subplg),
+                                    current_registry[plugin_type][interface][
+                                        plugin
+                                    ][subplg],
+                                )
+                        else:
+                            self.validate_plugin_attrs(
+                                plugin_type,
+                                interface,
+                                plugin,
+                                current_registry[plugin_type][interface][plugin],
+                            )
+                    except PluginRegistryError as e:
+                        print(e)
+                        pytest.xfail(str(e))
+
+    def validate_plugin_attrs(self, plugin_type, interface, name, plugin):
+        """Test non-product plugin for all required attributes."""
+        missing = []
+        if plugin_type == "yaml_based" and interface != "products":
+            attrs = [
+                "docstring",
+                "family",
+                "interface",
+                "package",
+                "plugin_type",
+                "relpath",
+            ]
+        elif plugin_type == "yaml_based":
+            attrs = [
+                "docstring",
+                "family",
+                "interface",
+                "package",
+                "plugin_type",
+                "product_defaults",
+                "source_names",
+                "relpath",
+            ]
+        else:
+            attrs = [
+                "docstring",
+                "family",
+                "interface",
+                "package",
+                "plugin_type",
+                "signature",
+                "relpath",
+            ]
+        for attr in attrs:
+            try:
+                plugin[attr]
+            except KeyError:
+                missing.append(attr)
+        if missing:
+            raise PluginRegistryError(
+                f"Plugin '{name}' is missing the following required "
+                f"top-level properties: '{missing}'"
+            )
+
+    def validate_registry_interfaces(self, current_registry):
+        """Test Plugin Registry interfaces validity."""
+        from geoips import interfaces
+        import inspect
+
+        yaml_interfaces = [
+            str(info[0]) for info in inspect.getmembers(
+                interfaces.yaml_based,
+                inspect.ismodule)
+        ]
+        module_interfaces = [
+            str(info[0]) for info in inspect.getmembers(
+                interfaces.module_based,
+                inspect.ismodule)
+        ]
```

Comment:
This is the replacement for listing all of the interfaces explicitly?  @jsolbrig probably has an idea on this as well.

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/commit/c48badcd01c9422a65b1b4a579e2d60163096091

There is the commit where it was updated

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/issues/397

Opened an issue to consider moving this to a function on geoips.interfaces, as @jsolbrig thinks this may not be the only place we would want this information.  I'm good resolving this now,  and we can move this functionality later if needed

Reply:
Sounds great! I agree that this may be useful information for other aspects of GeoIPS. I agree that it should probably be moved there in the future.